boards/st-stm32g0b1re-nucleo/HwInit.cxx

```diff
@@ -0,0 +1,328 @@
+/// care of that.
+HAL_StatusTypeDef HAL_InitTick(uint32_t TickPriority)
+{
+    return HAL_OK;
+}
+
+/** Initialize the processor hardware.
+ */
+void hw_preinit(void)
+{
+     /* Globally disables interrupts until the FreeRTOS scheduler is up. */
+    asm("cpsid i\n");
+
+    /* these FLASH settings enable opertion at 64 MHz */
+    __HAL_FLASH_PREFETCH_BUFFER_ENABLE();
+    __HAL_FLASH_SET_LATENCY(FLASH_LATENCY_2);
+
+    /* setup the system clock */
+    clock_setup();
+
+    /* enable peripheral clocks */
+    __HAL_RCC_GPIOA_CLK_ENABLE();
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+    __HAL_RCC_GPIOC_CLK_ENABLE();
+
+    __HAL_RCC_USART2_CLK_ENABLE();
+    __HAL_RCC_FDCAN_CLK_ENABLE();
+    __HAL_RCC_TIM17_CLK_ENABLE();
+
+    /* setup pinmux */
+    GPIO_InitTypeDef gpio_init;
+    memset(&gpio_init, 0, sizeof(gpio_init));
+
+    /* USART2 pinmux on PA2 and PA3 */
+    gpio_init.Mode =        GPIO_MODE_AF_PP;
+    gpio_init.Pull =        GPIO_NOPULL;
+    gpio_init.Speed =       GPIO_SPEED_FREQ_LOW;
+    gpio_init.Alternate =   GPIO_AF1_USART2;
+    gpio_init.Pin =         GPIO_PIN_2;
+    HAL_GPIO_Init(GPIOA, &gpio_init);
+    gpio_init.Pin =         GPIO_PIN_3;
+    HAL_GPIO_Init(GPIOA, &gpio_init);
+
+    /* CAN pinmux on PC4 and PC5 */
+    gpio_init.Mode =        GPIO_MODE_AF_PP;
+    gpio_init.Pull =        GPIO_NOPULL;
+    gpio_init.Speed =       GPIO_SPEED_FREQ_LOW;
+    gpio_init.Alternate =   GPIO_AF3_FDCAN1;
+    gpio_init.Pin =         GPIO_PIN_4;
+    HAL_GPIO_Init(GPIOC, &gpio_init);
+    gpio_init.Pin =         GPIO_PIN_5;
+    HAL_GPIO_Init(GPIOC, &gpio_init);
+
+    GpioInit::hw_init();
+
+    /* Initializes the blinker timer. */
+    TIM_HandleTypeDef TimHandle;
+    memset(&TimHandle, 0, sizeof(TimHandle));
+    TimHandle.Instance =            TIM17;
+    TimHandle.Init.Period =         configCPU_CLOCK_HZ / 10000 / 5;
+    TimHandle.Init.Prescaler =      10000;
+    TimHandle.Init.ClockDivision =  0;
+    TimHandle.Init.CounterMode =    TIM_COUNTERMODE_UP;
+    TimHandle.Init.RepetitionCounter = 0;
+    if (HAL_TIM_Base_Init(&TimHandle) != HAL_OK)
+    {
+        /* Initialization Error */
+        HASSERT(0);
+    }
+    if (HAL_TIM_Base_Start_IT(&TimHandle) != HAL_OK)
+    {
+        /* Starting Error */
+        HASSERT(0);
+    }
+    __HAL_DBGMCU_FREEZE_TIM17();
+    SetInterruptPriority(TIM17_FDCAN_IT1_IRQn, 0);
+    NVIC_EnableIRQ(TIM17_FDCAN_IT1_IRQn);
+}
+
+void hw_init(void) {
+}
+
+void uart2_lpuart2_interrupt_handler(void)
+{
+    Stm32Uart::interrupt_handler(1);
+}
+
+void uart3_4_5_6_lpuart1_interrupt_handler(void)
+{
+    Stm32Uart::interrupt_handler(2);
+}
+
+// timer17_fdcan_it1_interrupt_handler()
+//
+// the timer17/fdcan_it1 isr is invoked from the interrupt vector table.  It 
+// in turn calls each of the shared isr routines: timer17, fdcan1_it1 and 
+// fdcan2_it2.
+//
+
+void timer17_fdcan_it1_interrupt_handler(void)
+{
+    blinker_interrupt_handler();
+    // todo: Fix fdcan Instances...
```

Comment:
Minor annoyance, the commented out code is spaced kind of weird. Also, the todo tag is not a doxygen tag.

Reply:
I will update this after I make the MCAN port.